test/e2e/acl_logging.go

```diff
@@ -0,0 +1,204 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	knet "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	logSeverityNamespaceAnnotation = "k8s.ovn.org/acl-logging"
+	maxPokeRetries                 = 15
+	ovnControllerLogPath           = "/var/log/openvswitch/ovn-controller.log"
+	pokeInterval                   = 1 * time.Second
+)
+
+var _ = Describe("ACL Logging", func() {
+	const (
+		denyAllPolicyName       = "default-deny-all"
+		initialDenyACLSeverity  = "alert"
+		initialAllowACLSeverity = "notice"
+		namespacePrefix         = "acl-logging"
+		pokerPodIndex           = 0
+		pokedPodIndex           = 1
+	)
+
+	fr := framework.NewDefaultFramework(namespacePrefix)
+
+	var (
+		nsName string
+		pods   []v1.Pod
+	)
+
+	setNamespaceACLLogSeverity := func(namespaceToUpdate *v1.Namespace, desiredDenyLogLevel string, desiredAllowLogLevel string) error {
+		if namespaceToUpdate.ObjectMeta.Annotations == nil {
+			namespaceToUpdate.ObjectMeta.Annotations = map[string]string{}
+		}
+		By("updating the namespace's ACL logging severity")
+		updatedLogSeverity := fmt.Sprintf(`{ "deny": "%s", "allow": "%s" }`, desiredDenyLogLevel, desiredAllowLogLevel)
+		namespaceToUpdate.Annotations[logSeverityNamespaceAnnotation] = updatedLogSeverity
+
+		_, err := fr.ClientSet.CoreV1().Namespaces().Update(context.TODO(), namespaceToUpdate, metav1.UpdateOptions{})
+		return err
+	}
+
+	BeforeEach(func() {
```

Comment:
I don't think we should have multiple `BeforeEach` functions within a single `Describe` block Maybe condense them into a single `BeforeEach`?